### PR TITLE
Updated expo to SDK 28

### PR DESCRIPTION
**Pull request by kashishgrover**
Also updated react native and react for that. Using Node 9 is necessary with newer versions of RN. Also added yarn lockfile. :)

### CODE REVIEW

**Comment by kamleshchandnani** [comment]
Deploy preview for *leaf-ui* processing.

Built with commit 2df81e4ea2a98c68e759fe9556022d3c608f5281

https://app.netlify.com/sites/leaf-ui/deploys/5b48e30d3813f01334b25556

File: package.json [hunk 1]
```diff
@@ -72,9 +72,9 @@
     "npm-run-all": "4.1.2",
     "prop-types": "15.6.1",
     "raw-loader": "0.5.1",
-    "react": "16.2.0",
-    "react-dom": "16.2.0",
-    "react-native": "0.50.3",
+    "react": "16.3.1",
+    "react-dom": "16.3.1",
+    "react-native": "https://github.com/expo/react-native/archive/sdk-28.0.0.tar.gz",
```
Comment (kamleshchandnani) [inline package.json]: Why are we adding the packages from the github tar file? isn't it released?

Comment (kashishgrover) [reply package.json]: it is released, but this is as per https://blog.expo.io/expo-sdk-v28-0-0-is-now-available-f30e8253b530

**Comment by kashishgrover** [comment]
@kamleshchandnani i felt that with node upgrade we should try and shift towards yarn / npm 6. 😛  Can't we let it be as per user's choice?

**Comment by kashishgrover** [comment]
done :P

**Comment by kashishgrover** [comment]
😐I guess this should be okay now

File: package.json [hunk 2]
```diff
@@ -85,9 +85,9 @@
     "stylelint-processor-styled-components": "1.3.1"
   },
   "peerDependencies": {
-    "react": "^16.2.0",
-    "react-dom": "^16.2.0",
-    "react-native": "^0.50.3",
+    "react": "^16.3.1",
```
Comment (kamleshchandnani) [inline package.json]: Hotrod and rodimus are still on 16.2. So I would recommend not to add peerDependencies for react to be ^16.3.1

File: package.json [hunk 3]
```diff
@@ -72,9 +72,9 @@
     "npm-run-all": "4.1.2",
     "prop-types": "15.6.1",
     "raw-loader": "0.5.1",
-    "react": "16.2.0",
-    "react-dom": "16.2.0",
-    "react-native": "0.50.3",
+    "react": "16.3.1",
```
Comment (kamleshchandnani) [inline package.json]: Does the new expo version needs 16.3 explicitly? Isn't it compatible with 16.2?

Comment (kashishgrover) [reply package.json]: i shall check

**Pull request merged**